.gitignore

```diff
@@ -1,9 +1,12 @@
 *.pyc
 *.swp
+*~
 MANIFEST
 dist
 build
 .cache
 .coverage
 .tox
 .pytest_cache
+.mypy_cache
+.gitignore
```

Comment:
IMO only files generated by bits of project infrastructure should go in .gitignore. Anything related to your personal editor should go in a global gitignore: https://stackoverflow.com/questions/7335420/global-git-ignore

mypy_cache is from an external tool not part of the python-bugzilla build system so it doesn't go there either. and I don't understand what ignoring .gitignore is supposed to accomplish? how does it work?

More generally this should be submitted separately because it's orthogonal to the goal of this series. adding it mid review too just complicates things on my end